book/src/development/infrastructure/release.md

```diff
@@ -7,89 +7,89 @@ Clippy is released together with stable Rust releases. The dates for these
-$ git checkout -b backport_remerge
-$ git merge upstream/beta
-$ git diff  # This diff has to be empty, otherwise something with the remerge failed
-$ git push origin backport_remerge  # This can be pushed to your fork
+When a release needs to be done, `cargo test` will fail, if the versions in the
+`Cargo.toml` are not correct. During that sync, the versions need to be bumped.
+This is done by running:
+
+```
+$ cargo dev release bump_version
 ```
 
-After this, open a PR to the master branch. In this PR, the commit hash of the
-`HEAD` of the `beta` branch must exist. In addition to that, no files should be
-changed by this PR.
+This will increase the version number of each relevant `Cargo.toml` file. After
+that, just commit the updated files with:
 
-## Update the `beta` branch
+```bash
+# XX should be exchanged with the corresponding version
+$ git commit -m "Bump Clippy version -> 0.1.XX" **/*Cargo.toml
+```
+
+## Find the Clippy commit
 
-This step must be done **after** the PR of the previous step was merged.
+For both, updating the `beta` and the `stable` branch, the first step is to find
+the Clippy commit of the last Clippy sync done in the respective Rust branch.
 
-First, the Clippy commit of the `beta` branch of the Rust repository has to be
-determined.
+Running the following command will get the commit for the specified branch:
 
 ```bash
-# Assuming the current directory corresponds to the Rust repository
-$ git fetch upstream
-$ git checkout upstream/beta
-$ BETA_SHA=$(git log --oneline -- src/tools/clippy/ | grep -o "Merge commit '[a-f0-9]*' into .*" | head -1 | sed -e "s/Merge commit '\([a-f0-9]*\)' into .*/\1/g")
+$ SHA=$(cargo dev release commit /path/to/rust <branch>)
 ```
 
-After finding the Clippy commit, the `beta` branch in the Clippy repository can
-be updated.
+Where the `/path/to/rust` is a relative path to a Rust clone and the `<branch>`
+is one of `stable`, `beta`, or `master`.
+
+## Update the `beta` branch
+
+After getting the commit of the `beta` branch, the `beta` branch in the Clippy
+repository can be updated.
 
 ```bash
-# Assuming the current directory corresponds to the Clippy repository
 $ git checkout beta
-$ git reset --hard $BETA_SHA
+$ git reset --hard $SHA
 $ git push upstream beta
 ```
 
-## Find the Clippy commit
+## Update the `stable` branch
 
-The first step is to tag the Clippy commit, that is included in the stable Rust
-release. This commit can be found in the Rust repository.
+After getting the commit of the `stable` branch, the `stable` branch in the
+Clippy repository can be updated.
 
 ```bash
-# Assuming the current directory corresponds to the Rust repository
-$ git fetch upstream    # `upstream` is the `rust-lang/rust` remote
-$ git checkout 1.XX.0   # XX should be exchanged with the corresponding version
-$ SHA=$(git log --oneline -- src/tools/clippy/ | grep -o "Merge commit '[a-f0-9]*' into .*" | head -1 | sed -e "s/Merge commit '\([a-f0-9]*\)' into .*/\1/g")
+$ git checkout stable
```

Comment:
Do you have some kind of script that performs all of these steps?

Reply:
#12759 Will have all the `cargo dev` scripts that take care of this.